nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/DeleteHDFS.java

```diff
@@ -141,20 +141,33 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             }
 
             Map<String, String> attributes = Maps.newHashMapWithExpectedSize(2);
+            boolean foundMissingFile = false;
             for (Path path : pathList) {
                 attributes.put("filename", path.getName());
                 attributes.put("path", path.getParent().toString());
                 if (fileSystem.exists(path)) {
                     fileSystem.delete(path, context.getProperty(RECURSIVE).asBoolean());
+
                     if (!context.hasIncomingConnection()) {
                         flowFile = session.create();
+                        session.transfer(session.putAllAttributes(flowFile, attributes), REL_SUCCESS);
                     }
-                    session.transfer(session.putAllAttributes(flowFile, attributes), REL_SUCCESS);
+
                 } else {
                     getLogger().warn("File (" + path + ") does not exist");
+
                     if (!context.hasIncomingConnection()) {
                         flowFile = session.create();
+                        session.transfer(session.putAllAttributes(flowFile, attributes), REL_FAILURE);
                     }
+
+                }
+            }
+            if (context.hasIncomingConnection()) {
+                // TODO we only put the last path deleted, change the semantic of the processor?
+                if (!foundMissingFile) {
```

Comment:
Looks like 'foundMissingFile' is set to false above and then there is this if/else.  Perhaps this is incomplete?  Goes with the 'todo'?